custom_components/ttlock/services.py

```diff
@@ -86,18 +87,57 @@ def register(self) -> None:
             supports_response=SupportsResponse.OPTIONAL,
         )
 
-    def _get_coordinators(self, call: ServiceCall) -> list[LockUpdateCoordinator]:
+        self.hass.services.register(
+            DOMAIN,
+            SVC_LIST_PASSCODES,
+            self.handle_list_passcodes,
+            schema=vol.Schema(
+                {
+                    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+                }
+            ),
+            supports_response=SupportsResponse.ONLY,
+        )
+
+    def _get_coordinators(
+        self, call: ServiceCall
+    ) -> list[tuple[str, LockUpdateCoordinator]]:
```

Comment:
My assumption (given I haven't worked on the code in a while) is that there is a n:1 mapping between entities and coordinators (which is 1:1 between lock entities and coordinators).

Your change totally works (and solves the conflict based on name that I pointed out), but it doesn't really make promises about uniqueness in the return type, which would be nice to have.

Maybe something like this would work reasonably well:

```suggestion
    ) -> dict[str, LockUpdateCoordinator]:
```

and

```
return {
  entity_id: coordinator
  for ...
  if ...
}
```